### DISCUSSION TITLE
V2fly 是如何看待 wireguard outbound 的？

### DISCUSSION THREAD

**Discussion by hellodword** [v2fly/v2ray-core#2435] [Q&A]
我觉得对于有分流需求的场景来说，相比常规的 wireguard 实现，这么用起来非常灵活

- https://xtls.github.io/config/outbounds/wireguard.html
- https://github.com/v2fly/v2ray-core/issues/1746

**Comment by AkinoKaede** [comment]
In my view, WireGuard is a VPN tunnel, it is more used to create a virtual local area network, unfortunately V2Ray can not achieve the above functions at present..

**Reply by hellodword** [reply]
> In my view, WireGuard is a VPN tunnel, it is more used to create a virtual local area network, unfortunately V2Ray can not achieve the above functions at present..

You're right, but [`Router`](https://www.v2fly.org/en_US/v5/config/router.html) of V2Ray is powerful, I think it's very useful with a userspace WireGuard implementation[^1].

For example, V2Ray is often used to bypass the GFW, and redirecting the CN traffics to [Cloudflare WARP](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/) is an important practice about security. With WireGuard outbound, the configuration will be easier.

---

[^1]: https://github.com/XTLS/Xray-core/tree/82b09550d0596e7b2316f874d59c8e4131eb3c5a/proxy/wireguard
